core/src/main/scala/org/apache/spark/h2o/H2OContext.scala

```diff
@@ -255,7 +265,142 @@ object H2OContext extends Logging{
     * @return H2O Context
     */
   def getOrCreate(sc: SparkContext): H2OContext = {
-    getOrCreate(sc, new H2OConf(sc))
+    getOrCreate(sc, new H2OConf(sc))(SQLContext.getOrCreate(sc))
+  }
+
+  val defaultFieldNames = (i: Int) => "f" + i
```

Comment:
The code from here down should be located in `org.apache.spark.converts` package in order to keep h2o context clean. There are classes which takes care of converting various types such as RDD containing primitive type, RDD containing product class, dataframe to h2o frames and back.